File: leetcode/python/1539_kth_missing_positive_number.py
class Solution:
    def findKthPositive(self, arr: list[int], k: int) -> int:
        l = 0
        r = len(arr) - 1
        while l <= r:
            m = (l+r) // 2
            missing = arr[m] - (m+1)
            if missing < k:
                l = m+1
            else:
                r = m-1
        return l + k

s = Solution()
print(s.findKthPositive(arr = [2,3,4,7,11], k = 5))
print(s.findKthPositive(arr = [1,2,3], k = 5))
        

File: leetcode/c++/1539_kth_missing_positive_number.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int l {0};
        int r = arr.size() - 1;
        while (l <= r) {
            int m = l + (r - l) / 2;
            if (arr[m] - m - 1 < k) {
                l = m + 1;
            } else {
                r = m - 1;
            }
        }
        return l + k;
    }
};